core/src/main/java/org/apache/carbondata/core/memory/IntPointerBuffer.java

```diff
@@ -75,17 +75,17 @@ public int get(int rowId) {
   }
 
   public void loadToUnsafe() {
-    try {
-      pointerMemoryBlock =
-          UnsafeSortMemoryManager.allocateMemoryWithRetry(this.taskId, pointerBlock.length * 4);
+    pointerMemoryBlock =
+        UnsafeSortMemoryManager.INSTANCE.allocateMemory(this.taskId, pointerBlock.length * 4);
+    // pointerMemoryBlock it means sort storage memory manager does not have space to loaf pointer
+    // buffer in that case use pointerBlock
+    if (null != pointerMemoryBlock) {
```

Comment:
better to add a log  when pointerMemoryBlock == null

Reply:
adding log for not getting sort storage memory won't be a good idea as it will add overhead in case of huge data

Reply:
get your idea, but how do we know  we didn't get the sort storeage memory in fact cluster?